janus-idp/operator
# Recreate Backstage Pod on changes of ConfigMaps/Secrets configured in CR

## Description
This pull-request added following functionality:
- Adds label and annotation to ConfigMaps/Secrets configured in Backstage CR. 
- By default rhdh.redhat.com/ext-config-sync=true which allow controller watch ConfigMaps/Secrets 
- Enqueuing function calculates hash of the data, compares it with saved in the deployment's pod template annotation and calls reconcile in case they do not match (configuration changed)
- Since Pod template is changed k8s restarting it and reconciliation function saves new hash for the next cycle.

## Which issue(s) does this PR fix or relate to

https://github.com/janus-idp/operator/issues/236

## PR acceptance criteria

- [ ] Documentation TBD, scratched in the issue comments


## How to test changes / Special notes to the reviewer
- Create Backstage CR with configuration 
- Create ConfigMap(s)/Secret(s) referenced in CR
- Check if Operator added label rhdh.redhat.com/ext-config-sync=true and annotation rhdh.redhat.com/backstage-name=<CR name>
- Make sure Backstage instance is running
- Change anything in the data field of CMs/Secrets
- Make sure Backstage started recreating the Pod
- Make sure files/env vars with new content mounted to backstage container
